src/main/java/application/rest/v1/configmaps/KindActionMappingProcessor.java

```diff
@@ -0,0 +1,462 @@
+                       ", compApiVersion = " + compApiVersion);
+
+        boolean match = false;
+        if ( (apiVersion != null) && (!apiVersion.isEmpty()) &&
+             (compApiVersion != null) && (!compApiVersion.isEmpty()) ) {   
+            if (apiVersion.equals(compApiVersion)) { // group/version - exact match
+                match = true;
+            } else {
+                String grp_version[] = apiVersion.split("/");
+                String comp_grp_version[] = compApiVersion.split("/");
+
+                if ( (grp_version.length == 2) && (comp_grp_version.length == 2) ){ // group/version - wildcard match
+                    if (grp_version[0].equals(comp_grp_version[0]) || (grp_version[0].equals(WILDCARD)) )
+                        if (grp_version[1].equals(comp_grp_version[1]) || (grp_version[1].equals(WILDCARD)) ) {
+                            match = true;
+                        }
+                } else if ( (grp_version.length == 1) && (comp_grp_version.length == 1) ) { // verion only
+                    if (grp_version[0].equals(comp_grp_version[0]) || (grp_version[0].equals(WILDCARD)) ) {
+                        match = true;
+                    }
+                }
+            }
+        }
+
+        if (Logger.isExitEnabled()) 
+            Logger.log(className, "isApiVersionMatch", Logger.LogType.EXIT, "match = " + match);
+        return match;
+    }
+
+    /**
+     * Examine the mapping properites passed in
+     * 
+     * @param name
+     * @param subkind
+     * @param kind
+     * @return KSN, KS, KN, or K 
+     */
+    private int exemineMappingProperties(String name, String subkind, String kind) {
+        if (Logger.isEntryEnabled()) 
+            Logger.log(className, "exemineMappingProperties", Logger.LogType.ENTRY, 
+                   "(name, subkind, kind) = (" + name +", " + subkind + ", " + kind + ")");
+
+        int retVal = -1;
+        if ((kind != null) && !kind.isEmpty()) {
+            if ((subkind != null) && !subkind.isEmpty()) {
+                if ((name != null) && !name.isEmpty()) {
+                    retVal = KSN;
+                } else {
+                    retVal = KS;
+                }
+            } else if ((name != null) && !name.isEmpty()) {
+                retVal = KN;
+            } else {
+                retVal = K;
+            }
+        } else {
+            if (Logger.isErrorEnabled()) 
+                Logger.log(className, "exemineMappingProperties", Logger.LogType.ERROR, 
+                           "kind given is null or empty string");
+        }
+
+        if (Logger.isExitEnabled()) 
+                Logger.log(className, "exemineMappingProperties", Logger.LogType.EXIT, 
+                           "retVal = " + retVal);
+
+        return retVal;
+    }
+
+    /**
+     * Find a matched configmap defined in a given KAM 
+     * 
+     * @param cNumFields
```

Comment:
Suggestion:  add a description for each @param - reader has to keep jumping back and forth to where this method is called to figure out what each parameter is.